.github/workflows/test.yaml

```diff
@@ -0,0 +1,75 @@
+name: Tests
+
+on:
+  workflow_call:
+
+jobs:
+  lint-and-unit-test:
+    name: Lint and unit tests
+    runs-on: ubuntu-20.04
```

Comment:
This is also kind of limiting, what if the charm requires a different Ubuntu version?

Reply:
We'll probably have to parametrize that. I've identified some other special cases but we'll probably have to go charm to make the appropriate changes. Do we have this scenario already?

Reply:
Parametrized for both unit and integration tests. I've seen jenkins needs to run the tests on ubuntu 18. Thank you for your input